File: contests/lc/1975.MaximumMatrixSum/solution.py
from typing import List


class Solution:
    def maxMatrixSum(self, matrix: List[List[int]]) -> int:
        total = 0
        neg_count = 0
        min_abs = 10**18
        zero_found = False

        for row in matrix:
            for val in row:
                if val < 0:
                    neg_count += 1
                if val == 0:
                    zero_found = True
                abs_val = abs(val)
                total += abs_val
                if abs_val < min_abs:
                    min_abs = abs_val

        if neg_count % 2 == 1 and not zero_found:
            total -= 2 * min_abs
        return total

File: contests/lc/1975.MaximumMatrixSum/solution.go
package main

func maxMatrixSum(matrix [][]int) int64 {
	var sum int64
	negCount := 0
	minAbs := int64(1<<63 - 1)
	zeroFound := false

	for i := 0; i < len(matrix); i++ {
		row := matrix[i]
		for j := 0; j < len(row); j++ {
			val := row[j]
			if val < 0 {
				negCount++
			}
			if val == 0 {
				zeroFound = true
			}
			absVal := int64(val)
			if absVal < 0 {
				absVal = -absVal
			}
			sum += absVal
			if absVal < minAbs {
				minAbs = absVal
			}
		}
	}

	if negCount%2 == 1 && !zeroFound {
		sum -= 2 * minAbs
	}
	return sum
}
